### DISCUSSION TITLE
Confused by SSH-Agent.

### DISCUSSION THREAD

**Discussion by PDX-Dragon** [community/community#53947] [New to GitHub] [answered]
Hi Folks:
 
I've started with Gwen Faraday's _Git and GitHub for Beginners - Crash Course_ .  I liked her tutorial, but things went off the tracks for me when the topic of the SSH-Agent was brought up.  
 
Anything I try to do with SSH-Agent results in "unable to start ssh-agent service, error :1058."  
 
I'm on a Windows PC and I think she's working on an Apple computer.  
 
Her first sample repository section went fine, I like her instruction.  

Apparently GIT doesn't like my SSH key to be used by more than one repository.  Perhaps proper use of the SSH-Agent and SSH's config file addresses the issue, but I can't get her suggestions to work on my PC.  
 
Is there another tutorial I need to be looking at?  
 
Thanks
Larry

**Comment by D2Machine** [comment, marked as answer]
Hi 🙌🏻 
this error seems to mean that ssh-agent is installed but its service (on windows) hasn't been started.
You can check this by running in Windows PowerShell:

``` > Get-Service ssh-agent ```

And then check the output of status is not running.
``` 
Status   Name               DisplayName
------   ----               -----------
Stopped  ssh-agent          OpenSSH Authentication Agent
```
Then check that the service has been disabled by running
```
> Get-Service ssh-agent | Select StartType

StartType
---------
Disabled
```
I suggest setting the service to start manually. This means that as soon as you run ssh-agent, it'll start the service. You can do this through the Services GUI or you can run the command in admin mode:
```
 > Get-Service -Name ssh-agent | Set-Service -StartupType Manual
 ```
 And if you Get ```Set-Service: Service 'OpenSSH Authentication Agent (ssh-agent)' cannot be configured due to the following error: Access is denied.``` error you should use an **admin shell**
 
Alternatively, you can set it through the GUI :

>start menu --> type Services --> find the OpenSSH Authentication Agent service

![KwDuu](https://user-images.githubusercontent.com/87209891/235163636-dd658120-faa9-4f32-a834-36a07f51431b.png)

Reference: https://stackoverflow.com/questions/52113738/starting-ssh-agent-on-windows-10-fails-unable-to-start-ssh-agent-service-erro

**Reply by PDX-Dragon** [reply]
Thanks Mirlahiji:
 
Everything's working now.  
 
I'm still a bit confused.  I was able to open and use the first distro on GitHub from Gwen's tutorial, I have several others from previous attempts to learn GIT.   
 
For some reason I didn't have credentials for Gwen's second example distro, creating a local distro on my PC and them uploading it to an empty distro on GitHub.  

`git remote add origin git@github.com: ...`  
 
I appreciate your assistance.  

Thanks
Larry

**Reply by D2Machine** [reply]
I am very glad that your problem has been solved.
Regarding the topic you raised, I hope I understand it well. Have you already created a repository on github?
- If you haven't created one and you want to create a new repository at the same time as Pushing, use this command:
`git remote add origin git@github.com:User/UserRepo.git`
- And if you have already created a repository on GitHub, use this command:
`git remote set-url origin git@github.com:User/UserRepo.git`

**Comment by LalarrShoplink** [comment]
Good

**Comment by Andrianina7** [comment]
Comment dirigé le code SSH-Agent sous Android s'il vous plait!

**Comment by Rodagus** [comment]
Hola mi primera duda es sobre si alguien habla español y lo segundo
necesito ayuda con la clave ssh, no me autentica, intento comprobar y nada. porfavor que alguien me ayude

**Comment by Mykola-Samchuk** [comment]
when the command is executed: npm run deploy
I get an error:
Host key verification failed.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists

although all other commands are like: git push, git commit, git add,...
work correctly
